From ab21a0312fe2df754fd7a4cf4c45050233ee443d Mon Sep 17 00:00:00 2001
From: Ludovic Rousseau <ludovic.rousseau+github@gmail.com>
Date: Tue, 3 May 2011 11:12:13 +0200
Subject: [PATCH] Use const for read only buffers

Change the usb_bulk_write() and usb_interrupt_write() API to use
"const char *bytes" for read only buffers (data sent to the device).

See Debian bug #473814
"libusb: Please use const where appropriate in headers"
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=473814

[stuge: Also change actual function definitions]
---
 libusb/core.c |    4 ++--
 libusb/usb.h  |    6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/libusb/core.c b/libusb/core.c
index 3da15c6..04b1822 100644
--- a/libusb/core.c
+++ b/libusb/core.c
@@ -777,7 +777,7 @@ API_EXPORTED int usb_bulk_read(usb_dev_handle *dev, int ep, char *bytes,
 	return usb_bulk_io(dev, ep, bytes, size, timeout);
 }
 
-API_EXPORTED int usb_bulk_write(usb_dev_handle *dev, int ep, char *bytes,
+API_EXPORTED int usb_bulk_write(usb_dev_handle *dev, int ep, const char *bytes,
 	int size, int timeout)
 {
 	if (ep & USB_ENDPOINT_IN) {
@@ -821,7 +821,7 @@ API_EXPORTED int usb_interrupt_read(usb_dev_handle *dev, int ep, char *bytes,
 	return usb_interrupt_io(dev, ep, bytes, size, timeout);
 }
 
-API_EXPORTED int usb_interrupt_write(usb_dev_handle *dev, int ep, char *bytes,
+API_EXPORTED int usb_interrupt_write(usb_dev_handle *dev, int ep, const char *bytes,
 	int size, int timeout)
 {
 	if (ep & USB_ENDPOINT_IN) {
diff --git a/libusb/usb.h b/libusb/usb.h
index b7af749..84e730f 100644
--- a/libusb/usb.h
+++ b/libusb/usb.h
@@ -298,12 +298,12 @@ int usb_get_descriptor(usb_dev_handle *udev, unsigned char type,
 	unsigned char index, void *buf, int size);
 
 /* <arch>.c */
-int usb_bulk_write(usb_dev_handle *dev, int ep, char *bytes, int size,
+int usb_bulk_write(usb_dev_handle *dev, int ep, const char *bytes, int size,
 	int timeout);
 int usb_bulk_read(usb_dev_handle *dev, int ep, char *bytes, int size,
 	int timeout);
-int usb_interrupt_write(usb_dev_handle *dev, int ep, char *bytes, int size,
-        int timeout);
+int usb_interrupt_write(usb_dev_handle *dev, int ep, const char *bytes,
+	int size, int timeout);
 int usb_interrupt_read(usb_dev_handle *dev, int ep, char *bytes, int size,
         int timeout);
 int usb_control_msg(usb_dev_handle *dev, int requesttype, int request,
-- 
1.7.6.1

